sympy/geometry/point.py

```diff
@@ -172,28 +176,31 @@ def is_collinear(*args):
         False
 
         """
+        args = [Point.pointify(p) for p in args]
 
         # Coincident points are irrelevant; use only unique points.
-        args = list(set(args))
-        if not all(isinstance(p, Point) for p in args):
-            raise TypeError('Must pass only Point objects')
+        # XXX: the current workaround for issue #11238 relies on the points
+        # being in a specific order, so don't let `set` change the order
+        #args = list(set(args))
```

Comment:
It would be better to leave this as it is and XFAIL the test. Also, does #11554 fix this? Hopefully so, because when this PR is merged with the current master the following fails:

```python
x = 8*tan(13*pi/45)/(tan(13*pi/45) + sqrt(3))
y = (-8*sqrt(3)*tan(13*pi/45)**2 + 24*tan(13*pi/45))/ \
    (-3 + tan(13*pi/45)**2)
assert Line(Point(0, 0), Point(1, -sqrt(3))).contains(Point(x, y)) is True
```

Some orderings of the points works and some don't. I suspect it has to do with the fact that the ratio of `y/x` simplifies to `-sqrt(3)` but the coordinates themselves don't simplify enough for the routine to succeed.

The code you deleted -- using `is_scalar_multiple` works, however. Any reason not to keep it?